### PR TITLE
Juliu dev

### PR DESCRIPTION
**Pull request by ju-liu**
在这一组commit中，我们将原有的NodalBC_3D_vtp和NodalBC_3D_vtu的功能合并到了NodalBC这样一个通用的类中。将INodalBC这一个基类做成了纯头文件的格式。

这些大大简化了程序。

### CODE REVIEW

File: examples/fsi/preprocess_fsi.cpp
Comment (Syj-0525) [inline examples/fsi/preprocess_fsi.cpp:1]: "VEC_T::insert_end( meshdir_file_list, sur_f_file_out )" in 423 row should be changed to "VEC_T::insert_end( meshdir_file_list, sur_f_file_in ) "

**Review by Syj-0525 — COMMENTED** [review]
For the element type 501 and element type 502 in the CFD code and the multi-inlet case in the fsi code, I have tested the modified code using different cores and compared it with the master branch before the modification, and the generated h5 files are the consistent, and the residuals for each step outputted at the time the program is run are also the same.

File: examples/fsi/preprocess_fsi.cpp
Comment (ju-liu) [reply examples/fsi/preprocess_fsi.cpp:1]: ok checkout the newest commit. Confirm if this is correct. @Syj-0525

**Review by Syj-0525 — COMMENTED** [review]
The modified code is correct after testing.

**Review by Syj-0525 — APPROVED** [review]

**Pull request merged**